clients/src/main/java/org/apache/kafka/clients/consumer/CommitFailedException.java

```diff
@@ -28,7 +28,12 @@
 
     private static final long serialVersionUID = 1L;
 
-    public CommitFailedException(String message) {
-        super(message);
+    public CommitFailedException() {
+        super("Commit cannot be completed since the group has already " +
+                "rebalanced and assigned the partitions to another member. This means that the time " +
```

Comment:
Ah right, I was getting here via the references to `CommitFailedException` and there's one where we check the generation using `AbstractCoordinator.generation()` which returns null for all non-`STABLE` states, but this works because you handle the `onJoinPrepare` before setting the state to `REBALANCING`.